types/types.go

```diff
@@ -980,7 +980,28 @@ func UnmarshalToNodeTags(s string) ([]string, error) {
 	return res, nil
 }
 
+func isBDF(addr string) bool {
+	bdfFormat := "[a-f0-9]{4}:[a-f0-9]{2}:[a-f0-9]{2}\\.[a-f0-9]{1}"
+	bdfPattern := regexp.MustCompile(bdfFormat)
+	return bdfPattern.MatchString(addr)
+}
+
 func CreateDefaultDisk(dataPath string, storageReservedPercentage int64) (map[string]longhorn.DiskSpec, error) {
+	if isBDF(dataPath) {
```

Comment:
And in the below block disk creation, we need to use driver `DiskDriverAuto` rather than `DiskDriverNone`

Reply:
> Before creating a block disk as the default disk, should we check if v2 data engine is enabled?
> If NO, should we error out here? Notice that this function erroring out will fail > KubernetesNodeController.syncKubernetesNode()

Added a check, thank you.

> And in the below block disk creation, we need to use driver DiskDriverAuto rather than DiskDriverNone

For block-type disk, we use`DiskDriverAuto`. Use `DiskDriverNone` for filesystem-type instead.